README.md

```diff
@@ -1,17 +1,261 @@
+
+      var values = $(this).serialize();
+
+      var posting = $.post('/posts', values);
+
+      posting.done(function(data) {
+        // TODO: handle response
+      });
+    });
+  });
+</script>
+```
+
+In here, we're making use of the very handy [jQuery `serialize()`
+method](http://api.jquery.com/serialize/), which takes our form data and
+serializes it for us. Sure, we could individually select each
+form element and build our own JSON data, but we've done enough
+of that in this unit, so we've earned this shortcut!
+
+Next we use jQuery `post()`, much like we've been using `get()` to
+retrieve data. We pass it the URL and our `values`.
+
+Finally, we're using the `posting` object to specify what should happen
+when the AJAX request is `done`. This is where we'll need to handle the
+response.
+
+**Advanced:** The jQuery `post()` method returns a [jqXHR](http://api.jquery.com/jQuery.ajax/#jqXHR) object, which we're storing in our `posting` variable. These `jqXHR` objects implement the [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) interface, which is used for deferred or asynchronous operations.
+
+If we reload our form and try to create a blog post... nothing happens.
+
+OR DOES IT?
+
+![hackers acid burn](http://i.giphy.com/MyfjuvdTI3jpu.gif)
+
+If we look at our running Rails server, we'll probably see something
+like this:
+
+```erb
+Started POST "/posts" for ::1 at 2016-02-23 01:48:39 -0500
+Processing by PostsController#create as */*
+  Parameters: {"utf8"=>"✓", "authenticity_token"=>"k2thnauWV/3diGaOesdnSZaygMsA7tR7N0qa/HVk4M2IO4Q0HCuOnTF43NgaAZr/2N/5iHyKu6boAB/fVb0sww==", "post"=>{"title"=>"123abc", "description"=>"easy as"}}
+```
+
+That looks just like a regular POST with regular `params`. Since Rails
+does the work of creating a `params` hash for us, it doesn't matter if
+it comes from the form or from an AJAX request.
+
+Okay, so we did create a post. But if we look in our controller, we're
+supposedly redirecting to our `post_path`. And if we look further into
+our running Rails server, we'll see that it *did* redirect after
+creating our post. What gives?
+
+### Getting a Response from an AJAX POST
+
+When we do an AJAX request, we're expecting data to be returned so that
+we can deal with it on the client side. So when our controller
+redirected and rendered the post `show` page, it actually sent that data
+to the `.done()` method.
+
+You can see this in action by adding a `console.log(data)` inside the
+`.done()` function and then inspecting the response.
+
+```javascript
+//...
+  posting.done(function(data) {
+    // TODO: handle response
+    console.log(data);
+  });
+//...
+```
+
+You'll see the full HTML of the `show` page for that new post.
+
+#### A Side-note About HTTP Codes
+
+HTTP status codes exist to let the client know what kind of response
+they are getting, and what to do with it.
+
+The code for a redirect is `302`. When a browser makes a request, and
```

Comment:
Should we let students know that `302` is what we use for this _kind_ of redirect?